### PR TITLE
[PORT] Replaces downloading research notes with a new antag objective and station sabotage point.

### PR DESCRIPTION
**Pull request by ToasterBiome**
# Document the changes in your pull request

From https://github.com/tgstation/tgstation/pull/61685, paraphrasing (and a lot of copypasting)

Removes the objective to download research notes.

Replaces it with an objective to steal the source code for Project BEE from the master R&D server in order to stop NT from creating a wormhole to Beestation.

Adds the master R&D server to every map, replacing one of the existing servers. This server utilizes a couple of overlays which sit ontop of the basic server sprite to make it stand out more. The first overlay is just some hazard warning markings. The second is opening the hard drive access panel, which is to be distinguished from opening the maintenance panel.

image

The master r&d server contains a special hard drive
image

which can only be extracted by any antagonist or special_role following the prescribed steps
image

Removing this hard drive or destroying the master r&d server cuts all research point generation in half. This cannot be undone and there is no way to make another master server, nor any way to insert a retrieved hard drive back into a master server.

It is expected this objective should take about a minute to complete from first tool use to getting the objective item in your hands.

image

There are multiple steps that auto-complete, making the appropriate noise with each iteration (unscrew, crowbar, snip wire). These can be completed all at once or parts can be completed bit-by-bit and returned to later if you're interrupted.

image

This new master server cannot be deconstructed by ordinary methods.

Space Ninja & New Objective
The space ninja has had an overhaul to the downloading research notes objective. It has been removed and they can no longer hack the R&D console. Instead, they can use their gloves to hack the master R&D server to wipe the data on the traitor objective HDD (If it is present. This will destroy the objective!) or on any R&D server (master without HDD or normal) to wipe all accrued research points, setting them to 0.

# Why is this good for the game?
The download research notes objective is hands down our least interesting, most basic steal objective that has 0 impact on the shift and 0 impact on the station or crew - tied pretty much with stealing an unused slime extract. There's absolutely no incentive to stop it when you notice it, beyond getting in your valid hunt quota for the week.

This new master R&D server is sabotage-minded and gives a tangible penalty for allowing it to be destroyed or sabotaged.

Similarly, the ninja is designed with station sabotage in mind. Being able to sabotage research efforts through wiping research points and halfing research point generation for the rest of the shift fall squarely into that sabotage niche and their impact is far more wide-reaching than stealing research notes, as well as the objective being much riskier to pull off due to the secluded nature of the R&D server room and the AI alert when it is attempted.

These interactions are far more interesting than stealing the research notes.

# Testing
https://replays.yogstation.net/?roundid=50075#31818;195.47;124.09;2;ninawindia;0.0
Player was able to successfully remove it.

# Spriting

# Wiki Documentation

New objective: Steal the source code for Project Bee from the master R&D server mainframe.

# Changelog

:cl:  Timberpoes, ToasterBiome
rscadd: Adds new Master RND server, with a special hard drive that if removed, halves point generation.
rscadd: Add a new "Steal Master HDD" objective, where you must steal the hard drive and escape with it.
rscadd: Gives Ninja a different objective to sabotage research, and using their gloves on the server will drain all the points.
rscdel: Removes "Steal Research" objective as it's stupid easy. 
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
why no compile... L

**Comment by azzzertyy** [comment]
i think if the hard drive is found again it should be able to be put back in, just makes sense innit
also encourages actually catching the guy who stole it so i see no real reason it shouldnt be able to be put back in

**Comment by JohnFulpWillard** [comment]
based af

**Comment by Firewolf34** [comment]
Excellent PR quality. Love all the gifs and explanation and care that went into this!

My only tiny suggestion is to give the tools red accents, if they're specific to the objective.

**Review by cuackles — APPROVED** [review]

**Review by Aquizit — APPROVED** [review]
Map stuff is simple and looks fine.

**Comment by ToasterBiome** [comment]
> Excellent PR quality. Love all the gifs and explanation and care that went into this!
> 
> My only tiny suggestion is to give the tools red accents, if they're specific to the objective.

The tools are not. You procure them yourself.

**Comment by ToasterBiome** [comment]
> i think if the hard drive is found again it should be able to be put back in, just makes sense innit also encourages actually catching the guy who stole it so i see no real reason it shouldnt be able to be put back in

Feel free to make that addition if this gets merged

**Comment by MajManatee** [comment]
The 29 Runtimes in question

File: code/modules/ninja/suit/ninjaDrainAct.dm [hunk 1]
```diff
@@ -144,14 +152,20 @@ They *could* go in their appropriate files, but this is supposed to be modular
...
 
-	if(stored_research)
-		to_chat(H, span_notice("Copying files..."))
-		if(do_after(H, S.s_delay, src) && G.candrain && src)
-			stored_research.copy_research_to(S.stored_research)
-	to_chat(H, span_notice("Data analyzed. Process finished."))
+	if(!do_after(H, 30 SECONDS, target = src))
+		return
+
+	SSresearch.science_tech.modify_points_all(0)
```
Comment (JohnFulpWillard) [inline code/modules/ninja/suit/ninjaDrainAct.dm]: Can you make this instead take modify the techweb the server in question is connected to

Comment (ToasterBiome) [reply code/modules/ninja/suit/ninjaDrainAct.dm]: > Can you make this instead take modify the techweb the server in question is connected to

yeah

Comment (JohnFulpWillard) [reply code/modules/ninja/suit/ninjaDrainAct.dm]: just replace ``SSreseach.science_tech`` with ``stored_research``

**Review by JohnFulpWillard — APPROVED** [review]

**Pull request merged**